lib/cc/config/prepare.rb

```diff
@@ -0,0 +1,68 @@
+module CC
+  module Config
+    class Prepare
+      class NoPrepareNeeded
+        def fetch
+          []
+        end
+      end
+
+      attr_reader :fetch
+
+      def self.from_yaml(data)
+        if data.present?
+          fetch = Fetch.from_yaml(data.fetch("fetch", []))
+
+          new(fetch: fetch)
+        else
+          NoPrepareNeeded.new
+        end
+      end
+
+      def initialize(fetch:)
+        @fetch = fetch
+      end
+
+      class Fetch
+        def self.from_yaml(data)
+          new(data.map { |d| Entry.from_yaml(d) })
+        end
+
+        def initialize(entries)
+          @entries = entries
+        end
+
+        def each(&block)
+          entries.each(&block)
+        end
+
+        private
+
+        attr_reader :entries
+
+        class Entry
+          attr_reader :url, :path
+
+          def self.from_yaml(data)
+            case data
+            when String then new(data)
+            when Hash then new(data.fetch("url"), data["path"])
+            end
+          end
+
+          def initialize(url, path = nil)
+            @url = url
+            @path = path || url.split("/").last
```

Comment:
Nice catch. I'd like to do a few things to solve this (only one of which will be in this PR):

- Merge https://github.com/codeclimate/codeclimate/pull/638 first

  This'll ensure `validate-config` would catch this with a nice user-facing error.

- Don't bring `prepare` back into Builder without `validate-config`

  This'll ensure we don't get to this step without having validated already.

- Duplicate that check here, for *layered security*, just with a less user-friendly outcome because we don't need to be super graceful in an unexpected case where `validate-config` was not run.